CRM/SMS/Form/Schedule.php

```diff
@@ -144,21 +152,20 @@ public function postProcess() {
       CRM_Core_Error::fatal(ts('Could not find a mailing id'));
     }
 
-    foreach (array('now', 'start_date', 'start_date_time') as $parameter) {
+    foreach (array('send_option', 'start_date') as $parameter) {
       $params[$parameter] = $this->controller->exportValue($this->_name, $parameter);
     }
 
-    if ($params['now']) {
+    if (isset($params['send_option']) && $params['send_option'] == 'send_immediate') {
       $params['scheduled_date'] = date('YmdHis');
     }
     else {
-      $params['scheduled_date'] = CRM_Utils_Date::processDate($params['start_date'] . ' ' . $params['start_date_time']);
+      $params['scheduled_date'] = CRM_Utils_Date::processDate($params['start_date']);
```

Comment:
@JKingsnorth @mattwire do we need processDate here either?

Reply:
Well spotted... no we shouldn't